flink-streaming-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraTupleWriteAheadSink.java

```diff
@@ -110,11 +92,25 @@ public void close() throws Exception {
 	}
 
 	@Override
-	protected void sendValues(Iterable<IN> values, long timestamp) throws Exception {
-		//verify that no query failed until now
-		if (exception != null) {
-			throw new Exception(exception);
-		}
+	protected boolean sendValues(Iterable<IN> values, long timestamp) throws Exception {
+		int updatesSent = 0;
+		final AtomicInteger updatesConfirmed = new AtomicInteger(0);
+
+		final AtomicContainer<Throwable> exception = new AtomicContainer<>();
```

Comment:
The Java library provides an `AtomicReference` implementation which we could use instead of `AtomicContainer`.